UltraStar Play/Assets/StreamingAssets/Mods/BatchIsolateVocals/1-BatchIsolateVocalsModSettings.cs

```diff
@@ -152,13 +153,33 @@ public VisualElement CreateVisualElement()
                 toggle.label = SongMetaUtils.GetArtistDashTitle(songMeta);
                 toggle.value = false;
                 toggle.userData = songMeta;
-                
-                toggles.Add(toggle);
 
+                if (!IsSongSupported(songMeta))
+                {
+                    toggle.SetEnabled(false);
+                }
+                toggles.Add(toggle);
                 toggleContainer.Add(toggle);
             }
 
             return toggleContainer;
         }
+
+        private readonly string[] supportedExtensions = { "wav", "mp3", "ogg", "m4a", "wma", "flac" };
```

Comment:
file extensions tend to not be a reliable indicator of what is in them, especially for media files. Also, supported file types and file formats depends on platform where the game is running on.
How about instead of filtering based on file type, maybe handle whatever error may happen during scanning, by adding some try catch?

Reply:
Current implementation for Vocals Isolation uses [SpleeterMsvcExe](https://github.com/wudicgi/SpleeterMsvcExe), which uses ffmpeg. I am unsure what file types are supported there.
Anyway, the supported file types for Vocals Isolation may differ from what the game supports for playback.

The approach with file extensions is probably "good enough" for most cases. I would be fine with that.
However, the better solution would be more forgiving error handling. But it requires adjustments of the current job system. I can take a look at this soon.

Reply:
This change does not make anything worse, does it? What is our standard of quality for contributions?

Reply:
As I said, I am fine with this. It is only for a mod anyway.